tests/unit/Http/SetCookieTest.php

```diff
@@ -411,7 +411,7 @@ public function testParseCookie($cookie, array $parsed): void
 
             if (isset($p['Expires'])) {
                 // Remove expires values from the assertion if they are relatively equal
-                if (\abs($p['Expires'] != \strtotime((string) $parsed['Expires'])) < 40) {
+                if (\abs((int) ($p['Expires'] != \strtotime((string) $parsed['Expires']))) < 40) {
```

Comment:
Note for me: Take a look what upstream (Guzzle) to fix the test.

Reply:
Should be fine.

Reply:
I don't understand this if statement, maybe you or @sualko can help.

Here are some thoughts:

$p['Expires'] is an integer timestamp e.g. 1217091642
$parsed['Expires'] is a time string oder integer timestamp e.g. Sat, 26-Jul-2008 17:00:42 GMT or 1217091642

Converting the time string to a timestamp makes sense to me, even though it will fail for the case that the time string is a timestamp!
Comparing if they are not equal, will result in a boolean.
Why convert a boolean to an int (resulting in either 0 or 1) and getting the absolute value of it (will stay 0 or 1)? Why checking if lower than 40, if it can only be 0 or 1?

---

If I would want to check if the two cookie expire dates are 'relatively equal' I would substract one timestamp from the other, get the absolute value and check if lower than 40. 

e.g. `if (\abs((int) ($p['Expires'] - \strtotime((string) $parsed['Expires']))) < 40) {`

or, if you want to support the timestamp value in $parsed['Expires'] and expect $p['Expires'] to always be an integer: 

`if (\abs(($p['Expires'] - (is_int($parsed['Expires']) ? $parsed['Expires']: \strtotime($parsed['Expires'])))) < 40) {`

Reply:
Hm, yeah you are right. I was a bit confused how this `if` exactly was meant to work. Your proposals looks sensible to me.

Reply:
@sualko what do you think?

Reply:
I agree that that line doesn't make any sense. Additionally to Samuels suggestion, I would love to see some validation of the `strtotime` return value.

Reply:
I would prefer to skip this test with `self::markTestSkipped` for now. As this is not real our code and only taken from Guzzle we should wait for an upstream fix. If I am reading the PR for https://github.com/guzzle/guzzle/issues/2894 correctly, a lot of fixes are needed to make the test worked as expected again.

Reply:
I would be fine with that, as we are mententioned in the guzzle issue we should receive a notification once it is fixed

Reply:
Implemented as proposed: Original code restored and test will be starting from PHP 8 including a reference to the discussion here and the opened Guzzle issue.